lib/ansible/plugins/strategy/linear.py

```diff
@@ -52,6 +52,36 @@
 
 class StrategyModule(StrategyBase):
 
+    noop_task = None
+
+    def _replace_with_noop(self, target):
+        if self.noop_task is None:
+            raise AssertionError('strategy.linear.StrategyModule.noop_task is None, need Task()')
+
+        result = []
+        for el in target:
+            if isinstance(el, Task):
+                result.append(self.noop_task)
+            elif isinstance(el, Block):
+                result.append(self._create_noop_block_from(el, el))
```

Comment:
This line should be `el, el._parent`.  Otherwise, you are making the block, have a parent of the el, so we create an extra level of parent relationship.

Current:

```
       el: BLOCK(uuid=ca00b048-ca01-4d89-34fa-000000000079)(id=4442932392)(parent=TASK: role1 : call role2 again)
new_block: BLOCK(uuid=ca00b048-ca01-4d89-34fa-00000000011b)(id=4442932896)(parent=BLOCK(uuid=ca00b048-ca01-4d89-34fa-000000000079)(id=4442932392)(parent=TASK: role1 : call role2 again))
```

using `el._parent`:

```
       el: BLOCK(uuid=ca00b048-ca01-b999-e740-000000000079)(id=4602644128)(parent=TASK: role1 : call role2 again)
new_block: BLOCK(uuid=ca00b048-ca01-b999-e740-00000000011b)(id=4602644688)(parent=TASK: role1 : call role2 again)
```

Reply:
Good catch, thanks. Fixed.